### PR TITLE
Add Analytics test and update the new tests to incorporate mockgoose.js

### LINKED ISSUE #152: Add more unit tests
**Is your feature request related to a problem? Please describe.**
As of now, there are only the unit tests for auth routes and project routes.

**Describe the solution you'd like**
There should be tests for each route.

**Describe alternatives you've considered**
Implement unit tests using Mocha and chai.js

**Additional context**
It will improve the quality of the code.

### PR DESCRIPTION
**Pull request by carpecodeum**
# Description

Add analytics tests and updated the new tests to incorporate mockgoose.js

Fixes #152 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Screenshot 2020-02-25 at 6 39 58 PM

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

Also, include screenshots of app for the verification and reviewing purpose.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by carpecodeum** [comment]
@UdeshUK please review

**Comment by carpecodeum** [comment]
@Tushar19varshney please review

File: labellab-server/test/analytics.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+const Info = require('./info')
+
+const userInfo = {
+	name: 'name',
+	username: 'username',
+	email: 'email@gmail.com',
+	password: 'password',
+	password2: 'password'
+}
+
```
Comment (Tushar19varshney) [inline labellab-server/test/analytics.js]: Instead of creating a new userInfo object you can use dump data from `info.js`.

**Comment by Tushar19varshney** [comment]
@carpecodeum After registering a test user, it would be really great if you can write a cleanup function for deleting the data you created for testing. It will create unnecessary users in dB.

**Comment by carpecodeum** [comment]
> @carpecodeum After registering a test user, it would be really great if you can write a cleanup function for deleting the data you created for testing. It will create unnecessary users in dB.

@Tushar19varshney yes that is why a mock db is created using mockgoose before running test instances and after the tests are done the user info and the test data gets deleted!

File: labellab-server/test/analytics.js (hunk 1 above)
Comment (carpecodeum) [reply labellab-server/test/analytics.js]: sure I will do that!

**Pull request merged**